### ISSUE TITLE
Whitelisting dependencies

### ISSUE THREAD

**Issue by cihati** [dependabot/feedback#311]
I am exploring dependabot for our private repository. What I'm trying to achieve:

1) Whitelist some dependencies that I want to keep updated
2) Some tool (in this case, hopefully dependabot) sends PR to upgrade package.json and yarn.lock (also, I'd ideally only want changes to both package.json and yarn.lock, not just yarn.lock)

Hopefully, over time I'll add more and more stuff to this whitelist but I do not want to go all-in from the beginning.

Is it possible with dependabot?

**Comment by greysteil** [comment]
Thanks for getting in touch @cihati.

> Whitelist some dependencies that I want to keep updated

For now this isn't possible with Dependabot. You can ask Dependabot to ignore dependencies, but only after it's created an initial PR for them - there's no way to add ignores in advance, or to whitelist (as opposed to blacklist). Support for both is coming in the next month as part of the config files work we're doing.

> Some tool (in this case, hopefully dependabot) sends PR to upgrade package.json and yarn.lock (also, I'd ideally only want changes to both package.json and yarn.lock, not just yarn.lock)

Dependabot can do this bit just fine. When you say "not just yarn.lock" do you mean not just dependencies that only appear in the lockfile, or not updates to versions that are already covered in the requirements in your `package.json`?

**Comment by cihati** [comment]
Got it -- then I'll probably start using dependabot next month once you have this feature.

What I mean is that whenever dependabot opens a PR, it should touch both package.json and yarn.lock (so I do *not* want changes that only affect yarn.lock but not package.json)

**Comment by greysteil** [comment]
Interesting, there's a chance that Greenkeeper might be a better fit for you in that case:
- You an configure Dependabot to update your `package.json` as well as your `yarn.lock` whenever it's making an update (so a `^1.0.1` requirement would get bumped to `^1.0.2` in your `package.json`, even though the new version was covered by the old requirement).
- What you *can't* currently do easily with Dependabot is ask is to only update both your lockfile and `package.json` when `2.0.0` comes out, ignoring all intermediate dependencies. (You can do this on a dependency-by-dependency basis by configuring ignore conditions, but can't set it as a general update strategy.)
- Greenkeeper, on the other hand, was designed to work with libraries rather than apps, where behaviour like the above is perhaps closer to what you want. As I understand it (from reading their docs a while back - I haven't used their product):
  - They'll alert you if a new release in the `1.x.x` series breaks your build, but won't open PRs for these
  - They'll open a PR to update your requirement when it no longer covers the latest version.

Is that the behaviour you're looking for, or am I missing something?

**Comment by cihati** [comment]
> What you _can't_ currently do easily with Dependabot is ask is to only update both your lockfile and package.json when 2.0.0 comes out, ignoring all intermediate dependencies

what do you mean by "ignoring all intermediate dependencies"? I am fine with Dependabot upgrading other dependencies of the main package being upgraded (i.e. the one in package.json), so they can be in yarn.lock

Also, on a seperate topic:
What version of yarn does dependabot use? is it possible to specify it? because the generated yarn.lock differs from version to version sometimes (i.e. one version does not have SHA other one has it etc)

**Comment by greysteil** [comment]
Sorry, but "intermediate dependencies I meant intermediate updates. So in this example, 1.0.2 would be an intermediate update - would you want that ignored (i.e., no PR for it) or would you want a PR that updates both the `package.json` and the `yarn.lock` for it?

> What version of yarn does dependabot use?

We use 1.12.3. We have a bunch of code to clean up the lockfile and ensure there are no style changes, though 🙂

**Comment by cihati** [comment]
@greysteil Ideally, I'd want a PR that updates both package.json and the yarn.lock for that upgrade, yes

**Comment by greysteil** [comment]
Ah cool, Dependabot can handle that just fine. Thought you were asking for something weird for a minute there - keeping the `package.json` and `yarn.lock` in sync is easy. 🙂

**Comment by greysteil** [comment]
Whitelisting is now possible! 🎉 

It has to be done through a config file rather than your dashboard (we'll probably extend it to the dashboard in future), and the docs are [here](https://github.com/dependabot/feedback/issues/70#issuecomment-449377695).

Config files are brand new, so if anything's unclear just let us know. We're working to improve them, and to add more documentation and linting for them.
